File: factory/php/NotifyFactory.php
<?php

namespace Factory;

use Factory\INotify;
use Factory\MailNotify;
use Factory\SmsNotify;

class NotifyFactory
{
    public function createNotify(string $notifyType)
    {
        if ($notifyType == "SMS")
        {
            // Buradaki nesne oluşturma süreçleri bize kalmıştır.
            // Kullanıcıdan soyutlanmıştır.
            // Örnek olarak burası Singleton olarak da ayarlanabilirdi.
            return new SmsNotify();
        }
        else if ($notifyType == "MAIL")
        {
            // Buradaki nesne oluşturma süreçleri bize kalmıştır.
            // Kullanıcıdan soyutlanmıştır.
            // Örnek olarak burası Singleton olarak da ayarlanabilirdi.
            return new MailNotify();
        }
        return null;
    }
}

File: factory/python/notify_factory.py
# -*- coding: utf-8 -*-

from inotify import INotify
from sms_notify import SmsNotify
from mail_notify import MailNotify

class NotifyFactory:
    def create_notify(self, notify_type: str) -> INotify:
        if notify_type == 'SMS':
            # Buradaki nesne oluşturma süreçleri bize kalmıştır.
            # Kullanıcıdan soyutlanmıştır.
            # Örnek olarak burası Singleton olarak da ayarlanabilirdi.
            return SmsNotify()
        elif notify_type == 'MAIL':
            # Buradaki nesne oluşturma süreçleri bize kalmıştır.
            # Kullanıcıdan soyutlanmıştır.
            # Örnek olarak burası Singleton olarak da ayarlanabilirdi.
            return MailNotify()
        return None
